### DISCUSSION TITLE
Feature proposal: add Wireguard to Network/VPN/VPN type

### DISCUSSION THREAD

**Discussion by Mile-Lile** [linuxdeepin/developer-center#2857] [General | 常规讨论]
Since all major VPN providers support WireGuard, can you add it to VPN Type pattern. Since now deepin supports L2TP, PPTP, OpenVPN, StrongSwan, VPNC...
![Screenshot_dde-control-center_20220107165824](https://user-images.githubusercontent.com/12752816/148571442-f2ba577f-2a11-4007-a398-ea10c6c93d2c.png)

**Comment by Mile-Lile** [comment]
and there is also qrencode that is available on debian that makes easier WG import config [qrencode](https://packages.debian.org/search?keywords=qrencode)

**Comment by Mile-Lile** [comment]
@justforlxz

**Comment by Mile-Lile** [comment]
QRencode can be used to generate qr codes, even in a terminal/console using UTF8 characters.
The syntax is:


`qrencode -t ansiutf8 < client.conf`

This will generate a QR code that is readable by the mobile client(WireGuard has a user space implementation for mobile devices available via the WireGuard app - available for Android and iOS).

The advantage of this approach is that there is no need to transfer sensitive information via data channels that can potentially be compromised and there is no need for any additional software.

**Comment by Mile-Lile** [comment]
so, basically there is need for implementing both server and client options in DE...

**Comment by meiyixiang** [comment]
11/5000 
Hello, we have included your suggestion

**Comment by Mile-Lile** [comment]
thank you very much! appreciate...

**Comment by Mile-Lile** [comment]
just to add some proposals what options should GUI have... these screenshots are taken from ddwrt WG server...
![Screenshot_select-area_20220402082620](https://user-images.githubusercontent.com/12752816/161370137-d64162e9-1f08-463e-b1a0-2c96d2447e6f.png)
![Screenshot_select-area_20220402082707](https://user-images.githubusercontent.com/12752816/161370139-44cc6fa1-d8b9-4232-8e13-91ff42e948b9.png)
![Screenshot_select-area_20220402082728](https://user-images.githubusercontent.com/12752816/161370141-abc0b1ba-7591-4b71-95f5-77da265ae749.png)

**Comment by meiyixiang** [comment]
OK, I will feed back your suggestion to our product person in charge

**Comment by meiyixiang** [comment]
We have adopted this suggestion and scheduled to implement it, but it may take some time due to the large demand

**Comment by Mile-Lile** [comment]
@meiyixiang 
first of all thank you for considering this suggestion at all... deepin is the only distro that listens to users... and I saw what huge work your devs did in 20.5... just awesome... developing wireguard in ddwrt with all these features took about 4 years... you can see it [here](https://svn.dd-wrt.com/log/src/router/wireguard) and maybe your devs can use some from its code... ddwrt is linux based and uses iptables for example...
secondly, maybe it is not necessary to implement all features immediately... I am already using WG from CLI in deepin and it works... most users don't need server setup and policy based routing... server setup is for companies and advanced users...
benefits of using WG instead of ovpn is that WG is working in kernel space not in users space so it is much faster... I hope this will benefit to many users of deepin...
Thank you once again and cheers from Serbia (small country in south europe)...

**Comment by meiyixiang** [comment]
67/5000 
You are welcome. We have accepted the VPN proposal, but it may take some time to implement it, because there are a lot of demands from users. The implementation will not necessarily follow the style of your image.

**Reply by Mile-Lile** [reply]
I understand... and appreciate everything you do...

**Comment by meiyixiang** [comment]
25/5000 
Hope to have more users to give us valuable advice, let us do better

**Comment by Mile-Lile** [comment]
bump! Hi everybody, is there any progress on this?

**Reply by BLumia** [reply]
Currently it's on the backlog but not working on implementing such feature. Maybe will get implement **after** v23's first stable release.

**Comment by meiyixiang** [comment]
@BLumia 这个你看看，我认为他是不是需要支持一种新的VPN格式吗？

**Reply by BLumia** [reply]
是，需要支持 Wireguard 类型的 VPN。

**Comment by Mile-Lile** [comment]
adding this as a reference https://github.com/deepin-community/wireguard

**Comment by meiyixiang** [comment]
We will schedule this demand into the plan, first need to investigate, if it is implemented, it will be released after the V23 stable version.

**Comment by Mile-Lile** [comment]
Thx Gary for the info!
